src/main/java/persistence/entity/EntitySnapshot.java

```diff
@@ -0,0 +1,60 @@
+package persistence.entity;
+
+import domain.Snapshot;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class EntitySnapshot {
+    private final Map<Integer, Snapshot> snapshot;
+
+    public EntitySnapshot() {
+        this.snapshot = new ConcurrentHashMap<>();
+    }
+
+    public Snapshot getSnapshot(Integer hashCode) {
+        return snapshot.get(hashCode);
+    }
+
+    public Object getEntity(Integer hashCode) {
+        return getSnapshot(hashCode).getObject();
+    }
+
+    public <I> Snapshot save(Integer hashCode, I input, Object data) {
+        snapshot.put(hashCode, new Snapshot(input, data));
+
+        return getSnapshot(hashCode);
+    }
+
+    public boolean isEntityInSnapshot(Integer hashCode) {
+        return snapshot.containsKey(hashCode);
+    }
+
+    public int size() {
+        return snapshot.size();
+    }
+
+    public void clear() {
+        snapshot.clear();
+    }
+
+    public Map<Integer, Snapshot> exploreInSnapshot(EntityPersistenceContext context) {
```

Comment:
```suggestion
    public Map<Integer, Snapshot> exploreInSnapshot(EntityPersistenceContext entityPersistenceContext) {
```

이렇게하는게 더 이해하기 쉬울것같아요.